### PR TITLE
implemented getCommitByAuthor Endpoint

### PR DESCRIPTION
**Pull request by CVM0410**
Implemented getCommitByAuthor endpoint. Added test cases and added swagger annotation for the endpoint.

### CODE REVIEW

File: git-service
Comment (sarthakgoel1997) [inline git-service:1]: Please remove this binary file

File: pkg/handler/commit.go [hunk 1]
```diff
@@ -441,3 +442,74 @@ func GetCommitStatuses(owner, repo, commitSHA string) ([]Status, error) {
...
+	req, err := http.NewRequest(method, baseUrl, nil)
+	if err != nil {
+		http.Error(w, "Error generating new request: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Add authorization header if a personal access token is provided
+	if request.PersonalAccessToken != "" {
+		req.Header.Add("Authorization", "token "+request.PersonalAccessToken)
+	}
```
Comment (sarthakgoel1997) [inline pkg/handler/commit.go:469]: I guess 2 more headers need to be set for Auth. Please use an existing method for this part - `AddhttpAuthRequestHeaders` or `AddAuthRequestHeaders`

File: pkg/handler/commit.go [hunk 2]
```diff
@@ -441,3 +442,74 @@ func GetCommitStatuses(owner, repo, commitSHA string) ([]Status, error) {
...
+			http.Error(w, "Error reading response: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		var commits []model.CommitData
+		err = json.Unmarshal(body, &commits)
+		if err != nil {
+			http.Error(w, "Error unmarshalling JSON: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
```
Comment (sarthakgoel1997) [inline pkg/handler/commit.go]: I think there is a common util function for this part - `getCommitsByPageNumber`
We can consider using that

**Review by sarthakgoel1997 — CHANGES_REQUESTED** [review]

File: pkg/handler/commit.go (hunk 2 above)
Comment (CVM0410) [reply pkg/handler/commit.go]: Updated the code using getCommitsByPageNumber

File: pkg/handler/commit.go (hunk 1 above)
Comment (CVM0410) [reply pkg/handler/commit.go:469]: Using this is giving JSON unmarshalling error. Couldn't resolve it.

**Review by sarthakgoel1997 — APPROVED** [review]

**Pull request merged**